internal/build/lifecycle_execution.go

```diff
@@ -860,11 +851,11 @@ func (l *LifecycleExecution) hasExtensionsForRun() bool {
 }
 
 func (l *LifecycleExecution) runImageAfterExtensions() string {
-	var amd analyzedMD
+	var amd files.Analyzed
 	if _, err := toml.DecodeFile(filepath.Join(l.tmpDir, "analyzed.toml"), &amd); err != nil {
 		return l.opts.RunImage
 	}
-	if amd.RunImage == nil {
+	if amd.RunImage == nil || amd.RunImage.Image == "" {
```

Comment:
It looks like `github.com/BurntSushi/toml` had a bug before where empty structs without an `omitempty` annotation weren't getting written out to TOML files, so in our fixture file `run-image` was not present and `amd.RunImage` actually was `nil`. This change fixes the test no matter which version of the library we use.

cc @jkutner

Reply:
nice! thanks